eml/api/api.js

```diff
@@ -85,6 +121,78 @@ export const getAllComponents = async (components) => {
     components,
   };
   // Send request to S3 server
-  const res = await axios.post(url + '/api/component/getallcomponents', obj);
+  const res = await axios.post(url + '/api/components/all', obj);
   return res.data;
 };
+
+/*** COURSE ***/
+
+export const getCourses = async () => {
+  const res = await axios.get(url + '/api/courses')
```

Comment:
Again: singular words in URLs

Reply:
i get error if i use singular words and then "all" afterwards...

ex:

CastError: Cast to ObjectId failed for value "all" (type string) at path "_id" for model "course"

Reply:
I assume this is due to the backend using 'components' too. Both of these should be fixed.

Reply:
I am unsure if the problem is that the URL is in plural "courses" instead of singular "course" or if the problem is multiple words in "/getallcomponents/" instead of "/get-all-components/" or "/get/all/components/" but I have changed them so they are singular and without multiple words in both frontend(app) and backend